lib/ipmi_hpeoem.c

```diff
@@ -0,0 +1,430 @@
+    if (!rsp)
+    {
+        lprintf(LOG_ERR, "Error in getting nic selection");
+        return -1;
+    }
+    else if (rsp->ccode)
+    {
+        lprintf(LOG_ERR, "Error in getting nic selection (%s)",
+                val2str(rsp->ccode, completion_code_vals));
+        return -1;
+    }
+
+    nic_selection = rsp->data[3];
+    if (nic_selection <= 1)
+    {
+        nic_selection_str = ilo_nic_selection_mode[nic_selection];
+    }
+    printf("NIC selection: %s\n", nic_selection_str);
+
+    reset_needed = !!(rsp->data[4] & OEM_HPE_ILO_CONFIGURATION_STATUS_RESET_NEEDED_MASK);
+    printf("Reset needed: %s\n", reset_needed ? "yes" : "no");
+
+    // read shared nic information
+    if (nic_selection == OEM_HPE_ILO_SELECT_NIC_SHARED)
+    {
+        msg_data[1] = OEM_HPE_ILO_SHARED_NIC_SELECTION;
+        rsp = intf->sendrecv(intf, &req);
+        if (!rsp)
+        {
+            lprintf(LOG_ERR, "Error in getting nic selection");
+            return -1;
+        }
+        else if (rsp->ccode)
+        {
+            lprintf(LOG_ERR, "Error in getting nic selection (%s)",
+                    val2str(rsp->ccode, completion_code_vals));
+            return -1;
+        }
+
+        shared_nic_selection = rsp->data[1];
+        if (shared_nic_selection <= 2)
+        {
+            shared_nic_selection_str = ilo_shared_selection_mode[shared_nic_selection];
+        }
+        shared_nic_port = rsp->data[2];
+        if (shared_nic_port <= 2)
+        {
+            shared_nic_port_str = ilo_shared_nic_port[shared_nic_port];
+        }
+
+        printf("Shared NIC selection: %s\n", shared_nic_selection_str);
+        printf("Shared NIC port: %s\n", shared_nic_port_str);
+
+        printf("Shared NIC support:");
+        if (rsp->data[3] & OEM_HPE_ILO_SHARED_ALOM_SUPPORTED)
+        {
+            printf(" ALOM");
+        }
+        if (rsp->data[3] & OEM_HPE_ILO_SHARED_LOM_SUPPORTED)
+        {
+            printf(" LOM");
+        }
+        printf("\n");
+    }
+
+    return 0;
+}
+
+static int
+oem_hpe_parse_nic_selection_mode(char **argv)
+{
+    if (argv[current_arg] && !strcmp(argv[current_arg], "dedicated"))
+    {
+        return OEM_HPE_LAN_NIC_DEDICATED;
+    }
+
+    if (!argv[current_arg] || strcmp(argv[current_arg], "shared"))
+    {
+        return OEM_HPE_LAN_NIC_INVALID;
+    }
+    current_arg++;
+    if (!argv[current_arg] || !strcmp(argv[current_arg], "with"))
+    {
+        return OEM_HPE_LAN_NIC_INVALID;
+    }
+
+    current_arg++;
+    if (argv[current_arg] && !strcmp(argv[current_arg], "lom1"))
+    {
+        return OEM_HPE_LAN_NIC_SHARED_WITH_LOM1;
+    }
+    else if (argv[current_arg] && !strcmp(argv[current_arg], "lom2"))
+    {
+        return OEM_HPE_LAN_NIC_SHARED_WITH_LOM2;
+    }
+    else if (argv[current_arg] && !strcmp(argv[current_arg], "alom1"))
+    {
+        return OEM_HPE_LAN_NIC_SHARED_WITH_ALOM1;
+    }
+    else if (argv[current_arg] && !strcmp(argv[current_arg], "alom2"))
```

Comment:
It looks a lot like all those `if`/`else if` could be reduced to a simple dictionary and a loop:

```c
#define OEM_HPE_MAX_NIC_SIZE 10 /* the length of word 'dedicated' plus a null-byte */
struct elem {
    char type[OEM_HPE_MAX_NIC_SIZE];
    int rc;
    bool advance;
} dictionary = {
        { "dedicated", OEM_HPE_LAN_NIC_DEDICATED, false },
        { "shared", OEM_HPE_LAN_NIC_INVALID, true },
        { "with", OEM_HPE_LAN_NIC_INVALID, true },
        { "lom1", OEM_HPE_LAN_NIC_SHARED_WITH_LOM1, false },
        /*...*/
        { NULL, 0, false }
    };
    for (int i = 0; dictionary[i].name; ++i) {
        if (argv[0] && !strcmp(argv[0], dicttionary[i].name)) {
            return dictionary[i].rc;
        }
        argv += dictionary[i].advance ? 1 : 0;
    }
}
```
or something like that. Separate the code from the data.